charm-slurmd/src/interface_slurmd.py

```diff
@@ -158,12 +191,28 @@ def _store_slurmctld_host_port(self, host, port):
         if port != self._stored.slurmctld_port:
             self._stored.slurmctld_port = port
 
-    def get_stored_munge_key(self):
-        """Retrieve the munge_key from the StoredState."""
-        return self._stored.munge_key
-
     def _store_slurmd_restart_uuid(self, restart_slurmd_uuid):
         self._stored.restart_slurmd_uuid = restart_slurmd_uuid
 
     def _get_slurmd_restart_uuid(self):
         return self._stored.restart_slurmd_uuid
+
+    def get_stored_munge_key(self):
+        """Retrieve the munge_key from the StoredState."""
+        return self._stored.munge_key
+
+    def get_node_inventory(self):
+        """Retrieve the node inventory from the relation."""
+        return self._relation.data[self.model.unit].get("inventory")
```

Comment:
Do we need this method? We already have a `node_inventory` property.

Reply:
good point